lib/ansible/module_utils/urls.py

```diff
@@ -481,29 +483,37 @@ def redirect_request(self, req, fp, code, msg, hdrs, newurl):
             elif follow_redirects in ['no', 'none', False]:
                 raise urllib_error.HTTPError(newurl, code, msg, hdrs, fp)
 
+            method = req.get_method()
+
             do_redirect = False
             if follow_redirects in ['all', 'yes', True]:
                 do_redirect = (code >= 300 and code < 400)
 
             elif follow_redirects == 'safe':
-                m = req.get_method()
-                do_redirect = (code >= 300 and code < 400 and m in ('GET', 'HEAD'))
+                do_redirect = (code >= 300 and code < 400 and method in ('GET', 'HEAD'))
 
             if do_redirect:
-                # be conciliant with URIs containing a space
+                # Be conciliant with URIs containing a space
                 newurl = newurl.replace(' ', '%20')
                 newheaders = dict((k, v) for k, v in req.headers.items()
-                                  if k.lower() not in ("content-length", "content-type"))
+                                  if k.lower() not in ("content-length"))
+
                 try:
                     # Python 2-3.3
+                    data = req.get_data()
                     origin_req_host = req.get_origin_req_host()
                 except AttributeError:
                     # Python 3.4+
+                    data = req.data
                     origin_req_host = req.origin_req_host
-                return urllib_request.Request(newurl,
-                                              headers=newheaders,
-                                              origin_req_host=origin_req_host,
-                                              unverifiable=True)
+
+                return RequestWithMethod(newurl,
+                                         method=method,
+                                         headers=req.headers,
```

Comment:
Is there a reason we are dropping `newheaders` here?

Reply:
No probably a left-over of reworking the code. It works fine for our test-cases because there's no rewrite of the URL.